parties/githubclient.go

```diff
@@ -1,18 +1,229 @@
 package parties
 
 import (
+	"bytes"
+	"context"
 	"crypto/ecdsa"
+	"crypto/rsa"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/bastionzero/openpubkey/gq"
+	"github.com/bastionzero/openpubkey/pktoken"
+	"github.com/bastionzero/openpubkey/util"
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/lestrrat-go/jwx/v2/jws"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+	"github.com/zitadel/oidc/v2/pkg/client"
 )
 
+const githubIssuer = "https://token.actions.githubusercontent.com"
+
 type GithubOp struct {
+	rawTokenRequestURL    string
+	tokenRequestAuthToken string
+}
+
+var _ OpenIdProvider = &GithubOp{}
```

Comment:
This is a little trick to ensure that `*GithubOp` implements the `OpenIdProvider`. If it doesn't, we'll get a compiler error here. I added it because `GithubOp` wasn't actually being used in this project, so the stub implementation no longer implemented the interface. Arguably if it's being used we don't need this, but I do quite like having the error near the implementation rather than the usage. The `_` name is special, so no variable with that name is actually created.

Turns out I slightly misremembered the pattern to use here, a more common implementation is
```golang
var _ OpenIdProvider = (*GithubOp)(nil)
```
so that there is no actual instance being created. I will update.